python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -0,0 +1,597 @@
+                                           1.0 * self.config.overscanScale)
+
+        if self.config.doGenerateAmpDict:
+            expDict = dict()
+            for amp in exposure.getDetector():
+                expDict[amp.getName()] = exposure
+            return expDict
+        else:
+            return exposure
+
+    def addGain(self, ampData, gain):
+        ampArr = ampData.array
+        ampArr[:] = ampArr[:] / gain
+
+    def amplifierAddXGradient(self, ampData, start, end):
+        """Add a x-axis linear gradient to an amplifier's image data.
+
+         This method operates in the amplifier coordinate frame.
+
+        Parameters
+        ----------
+        ampData : `lsst.afw.image.ImageF`
+            Amplifier image to operate on.
+        start : `float`
+            Start value of the gradient (at y=0).
+        end : `float`
+            End value of the gradient (at y=ymax).
+        """
+        nPixX = ampData.getDimensions().getX()
+        ampArr = ampData.array
+        ampArr[:] = ampArr[:] + (np.interp(range(nPixX), (0, nPixX - 1), (start, end)).reshape(1, nPixX)
+                                 + np.zeros(ampData.getDimensions()).transpose())
+
+
+class RawMockLSST(IsrMockLSST):
+    """Generate a raw exposure suitable for ISR.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.isTrimmed = False
+        self.config.doGenerateImage = True
+        self.config.doGenerateAmpDict = False
+
+        # Add astro effects
+        self.config.doAddSky = True
+        self.config.doAddSource = True
+
+        # Add optical effects
+        self.config.doAddFringe = True
+
+        # Add instru effects
+        self.config.doAddParallelOverscan = True
+        self.config.doAddSerialOverscan = True
+        self.config.doAddCrosstalk = True
+        self.config.doAddBias = True
+        self.config.doAddDark = True
+
+        self.config.doAddFlat = True
+
+
+class CalibratedRawMockLSST(RawMockLSST):
+    """Generate a trimmed raw exposure.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+#        self.config.isTrimmed = True
+        self.config.doGenerateImage = True
+
+        self.config.doAddSky = True
+        self.config.doAddSource = True
+
+        self.config.doAddFringe = True
+
+        self.config.doAddParallelOverscan = False
+        self.config.doAddSerialOverscan = False
+        self.config.doAddCrosstalk = False
+        self.config.doAddBias = False
+        self.config.doAddDark = False
+        self.config.doAddGain = False
+        self.config.doAddFlat = False
+
+        self.config.biasLevel = 0.0
+        self.config.readNoise = 10.0
+
+
+class ReferenceMockLSST(IsrMockLSST):
+    """Parent class for those that make reference calibrations.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.isTrimmed = False
```

Comment:
Similar question here:  if these are to be the reference calibrations, I think those need to be trimmed (we don't retain the overscan after combination).  I think there also needs to be a factor applied to the readNoise for this class of mocks:  a combined bias is made by adding multiple calibrated bias frames, so the readNoise should drop by a factor of sqrt(N).  I'm not sure if we want to make N a config option, but having this parent class set `self.config.readNoise *= 0.25  # Assume combined calibrations are made with 16 inputs.` would be good.  I doubt this is done right in the original task.